### PR TITLE
Fix analysis of IOperation descendants for BoundDelegateCreationExpre…

### PR DESCRIPTION
**Pull request by mavasani**
…ssion

**Customer scenario**

Customer uses IOperation analyzers in their code base and see large number of false positives in code with delegate creation expressions taking lambda arguments. For example see https://github.com/dotnet/roslyn/issues/8884#issue-134680222

**Bugs this fixes:**

Fixes https://github.com/dotnet/roslyn/issues/8884#issue-134680222

**Workarounds, if any**

Customer has to suppress all IOperation analyzers in code containing delegate creation expressions OR disable these analyzers completely.

**Risk**

Low. We have taken same approach for bunch of other bound nodes that still require IOperation design.

**Performance impact**

Low.

**Is this a regression from a previous update?**
No.

**Root cause analysis:**

The current IOperation API implementation of BoundDelegateCreationExpression has a bunch of issues and needs to be redesigned. This is tracked by #8897 and will be addressed post 15.3. Meanwhile, to unblock analyzers on code containing delegate creation expressions with lambda arguments, this bound node has been switched to OperationKind.None with override for Children property.

**How was the bug found?**

Dogfooding/customer reports.

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @AlekseyTs @cston @dotnet/analyzer-ioperation for review

**Comment by cston** [comment]
LGTM

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IParameterReferenceExpression.cs [hunk 1]
```diff
@@ -192,6 +192,96 @@ Member Initializers(4): IPropertyInitializer (Property: System.Int32 Class.X { g
...
+            string source = @"
+using System;
+
+class Class
+{
+    public delegate void Delegate(int x, int y);
+
+    public void Method(Delegate d)
+    {
+        var a = /*<bind>*/new Delegate(Method2)/*</bind>*/;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IParameterReferenceExpression.cs]: >Method2 [](start = 39, length = 7)

Please add a test with an explicit receiver (a parameter reference, for example). We should get it in the output.

Comment (mavasani) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IParameterReferenceExpression.cs]: I added a note to #17588 about adding this unit test when implementing IOperation API for bound delegate creation expression.

---
In reply to: [115620033](https://github.com/dotnet/roslyn/pull/19307#discussion_r115620033) [](ancestors = 115620033)

**Comment by mavasani** [comment]
Tagging @MattGertz for ask mode approval.

**Pull request merged**